temporal-serviceclient/src/main/java/io/temporal/serviceclient/RpcRetryOptions.java

```diff
@@ -184,6 +206,19 @@ public Builder setMaximumInterval(Duration maximumInterval) {
       return this;
     }
 
+    /**
+     * Maximum amount of jitter to apply. 0.2 means that actual retry time can be +/- 20% of the
+     * calculated time. Set to 0 to disable jitter. Must be lower than 1. Default is 0.1.
+     */
+    public Builder setMaximumJitter(double maximumJitter) {
+      if (!Double.isFinite(maximumJitter) || maximumJitter < 0 || maximumJitter >= 1.0) {
```

Comment:
It looks like we will throw an exception on NaN. We should allow/support our "magic value" that means "default".

Reply:
I'm not convinced about this one.

This is similar to `backoffCoefficient`. Marker value for "not explicitly configured" is 0, which will be substituted by a default value of 2.0 in `validateBuildWithDefaults()`, but `setBackoffCoefficient()` wouldn't accept that marker 0 as a valid input. Same with `setInitialInterval()` and `setCongestionInitialInterval()`.

Now, I don't mind fixing all of these, but my feeling is that one is not expected to call `setX` methods uselessly on these builders. If you want to use default value for some property, then just don't call the setter. As far as I can see, the only reason to actually have those "not explicitly configured" markers is to properly handle merging in `setRetryOptions()`, which makes these an internal implementation detail, not something that users should be aware of.

Reply:
>but setBackoffCoefficient() wouldn't accept that marker 0 as a valid input.

This should be fixed.

> Now, I don't mind fixing all of these, but my feeling is that one is not expected to call setX methods uselessly on these builders.

With remote configurations, it's fairly standard and needed. Typically, people take remote values and call setters. So, there should be a value that means "default".